### PR TITLE
Added Remove-vRAReservationStorage.ps1

### PR DESCRIPTION
**Pull request by skunkie**
I suggest adding Remove-vRAReservationStorage (I used Set-vRAReservationStorage as a template), as I could not "uncheck" a storage from a reservation with Set-vRAReservationStorage, though it has a parameter "Enable" (its purpose is unclear).
 As far as I understand, the docs and tests for each cmdlet are generated with special tools. Please help with that if you consider this cmdlet worth a try.

### CODE REVIEW

**Comment by jonathanmedd** [comment]
@skunkie Thanks for the PR. We'll check it out soon - just in the middle of getting release 3.0.0 out at the moment. The docs are auto-built from the function help, but the tests for each function need to be manually written.

You''ll need to add a test for your function to this: https://github.com/jakkulabs/PowervRA/blob/master/tests/Test008-Reservation.Tests.ps1

We can help with the testing part if necessary.

File: src/Functions/Public/reservation-service/Remove-vRAReservationStorage.ps1 [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    The Id of the reservation
+
+    .PARAMETER Path
+    The storage path
+
+    .INPUTS
+    System.String.
+
+    .OUTPUTS
+    System.Management.Automation.PSObject
```
Comment (jonathanmedd) [inline src/Functions/Public/reservation-service/Remove-vRAReservationStorage.ps1]: Nothing gets output from this function so can you remove this

File: src/Functions/Public/reservation-service/Remove-vRAReservationStorage.ps1 [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    .EXAMPLE
+    Get-vRAReservation -Name "Reservation01" | Remove-vRAReservationStorage -Path "Datastore01"
+
+#>
+[CmdletBinding(SupportsShouldProcess,ConfirmImpact="High")][OutputType('System.Management.Automation.PSObject')]
```
Comment (jonathanmedd) [inline src/Functions/Public/reservation-service/Remove-vRAReservationStorage.ps1]: Nothing gets output from this function so can you remove:

[OutputType('System.Management.Automation.PSObject')]

**Comment by jonathanmedd** [comment]
@skunkie I found that it worked find when going from 3 Storage Paths listed in the Reservation down to 2, but not from 2 down to 1. I get an error:

{"errors":[{"code":10104,"message":"Data serialization error.","systemMessage":"Could not read message [acceptableTypes: [application/*+json;charset=UTF-8, application/json;charset=UTF-8]]","moreInfoUrl":null}]}

Which suggests something wrong with the JSON. I did a quick check over and the logic in the code and produced JSON would seem to be OK. Could you test the same scenario and see if you get the same thing? Possibly the JSON needs to be formatted differently if there is just one Storage Path.

Also might be worth taking a look at the below for how we approached removing a Network from the reservation.

https://github.com/jakkulabs/PowervRA/blob/master/src/Functions/Public/reservation-service/Remove-vRAReservationNetwork.ps1

**Comment by skunkie** [comment]
@jonathanmedd you are right, the approach in https://github.com/jakkulabs/PowervRA/blob/master/src/Functions/Public/reservation-service/Remove-vRAReservationNetwork.ps1 is much better, so I made Remove-vRAReservationStorage.ps1 based on it. Now it works for 3 and 2 storages as expected.
I assume we also need Get-vRAReservationStorage, without access to the web interface it is hard to tell which storages we have in reservations.

**Comment by jonathanmedd** [comment]
@skunkie Great job - have tested this successfully now, so happy to approve and merge.

You are more than welcome to make a Get-vRAReservationStorage for us :-)

**Pull request merged**